config/policy.go

```diff
@@ -565,8 +565,16 @@ func (p *Policy) Matches(requestURL url.URL) bool {
 	}
 
 	if p.Regex != "" {
-		re, err := regexp.Compile(p.Regex)
-		if err == nil && !re.MatchString(requestURL.String()) {
+		rawRE := p.Regex
+		if !strings.HasPrefix(rawRE, "^") {
+			rawRE = "^" + rawRE
+		}
+		if !strings.HasSuffix(rawRE, "$") {
+			rawRE += "$"
+		}
+
+		re, err := regexp.Compile(rawRE)
```

Comment:
can we move regex compilation out of hot path? it seems to be called every time authz is invoked.

Reply:
I don't like that Validate modifies the state of Policy, but I updated the code as you requested.

Reply:
for the record, I was never fan of how we have so many side effects in `Validate` but we would need to factor all of them out to be consistent.